src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/VulnerableCapability.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Protocol;
+using NuGet.VisualStudio.Internal.Contracts;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class VulnerableCapability : IVulnerable
+    {
+        private IReadOnlyList<PackageVulnerabilityMetadataContextInfo> _vulnerabilities = [];
+        public IReadOnlyList<PackageVulnerabilityMetadataContextInfo> Vulnerabilities
+        {
+            get => _vulnerabilities;
+            private set => _vulnerabilities = [.. value.OrderByDescending(v => v.Severity)];
+        }
+
+        public bool IsVulnerable => Vulnerabilities.Count > 0;
+
+        public PackageVulnerabilitySeverity VulnerabilityMaxSeverity
+        {
+            get
+            {
+                if (!IsVulnerable)
+                {
+                    throw new InvalidOperationException("Vulnerabilities is empty");
+                }
+
+                // Vulnerabilities are ordered on set so the first element is always the highest severity
+                int severity = Vulnerabilities[0].Severity;
+                if (Enum.IsDefined(typeof(PackageVulnerabilitySeverity), severity))
```

Comment:
I know that some `Enum` class static methods cause heap allocations on .NET Framework, in addition to bing relatively slow, which is why `AuditUtility.cs` doesn't use `Enum.ToString()` or `Enum.Parse`.  I assume that `Enum.IsDefined` has the same issue.

https://github.com/NuGet/NuGet.Client/pull/5361#discussion_r1295613063

Reply:
Thank you for pointing out a potential allocation. What do you suggest instead? We are making a conversion between an int with the existing contract class to an Enum. We want to make sure that we handle said Enum accurately. I look forward to your actionable, alternative suggestions.

Reply:
Thanks @zivkan! We chatted offline and we agreed to defer this for this now, but be aware of opportunity for avoid allocation as the this evolves. Thanks for taking the time to provide this detail!